src/invoiceNote/components/InvoiceNoteListPage.js

```diff
@@ -165,15 +172,35 @@ class InvoiceNoteListPage extends PureComponent<Props, State> {
   }
 
   componentDidUpdate(prevProps) {
-    const {location: {search: currentSearch}} = this.props;
-    const {location: {search: prevSearch}} = prevProps;
-    if(currentSearch !== prevSearch) {
-      const query = getUrlParams(currentSearch);
-      const page = query.page ? Number(query.page) : 1;
-
-      this.setState({activePage: page}, () => {
-        this.search();
-      });
+    const {location: {search: currentSearch}, userActiveServiceUnit} = this.props;
+    const {location: {search: prevSearch}, userActiveServiceUnit: prevUserActiveServiceUnit} = prevProps;
+
+    const handleSearch = () => {
+      this.setSearchValues();
+      this.search();
+    };
+
+    if (userActiveServiceUnit) {
+      if (userActiveServiceUnit !== prevUserActiveServiceUnit) {
+        if (!this._hasFetchedInvoiceNotes) { // No search has been done yet
+          handleSearch();
+          this._hasFetchedInvoiceNotes = true;
+        } else {
+          // Search again after changing user active service unit only if not explicitly setting the service unit filter
+          if (!currentSearch.includes('service_unit')) {
+            handleSearch();
+          }
+        }
+      } else {
+        if (!this._hasFetchedInvoiceNotes) { // No search has been done yet
```

Comment:
This has fetched xyz pattern was already fixed to other files, but still exists in this file. Please refactor this `if (userActiveServiceUnit !== prevUserActiveServiceUnit)` check to match src/contacts/components/ContactsListPage.js.

Reply:
Done